broker/query_plan.go

```diff
@@ -35,75 +46,138 @@ func (bpn *blockingPlanNodeImpl) Add(nodes ...common.BlockingPlanNode) {
 	bpn.children = append(bpn.children, nodes...)
 }
 
-// MergeNode is a blocking plan node that merges results from sub nodes
-type MergeNode struct {
+func NewMergeNode(agg common.AggType) common.MergeNode {
+	return &mergeNodeImpl{
+		aggType: agg,
+	}
+}
+
+type mergeNodeImpl struct {
 	blockingPlanNodeImpl
 	// MeasureType decides merge behaviour
-	AggType common.AggType
+	aggType common.AggType
 }
 
-func (mn *MergeNode) Run() (result queryCom.AQLQueryResult, err error) {
+func (mn *mergeNodeImpl) AggType() common.AggType {
+	return mn.aggType
+}
+
+func (mn *mergeNodeImpl) Run(ctx context.Context) (result queryCom.AQLQueryResult, err error) {
 	nChildren := len(mn.children)
 	// checks before fan out
-	if common.Avg == mn.AggType {
+	if common.Avg == mn.aggType {
 		if nChildren != 2 {
 			err = utils.StackError(nil, "Avg MergeNode should have 2 children")
 			return
 		}
-		lhs, ok := mn.children[0].(*MergeNode)
+		lhs, ok := mn.children[0].(common.MergeNode)
 		if !ok {
 			err = utils.StackError(nil, "LHS of avg node must be sum node")
 			return
 		}
-		if common.Sum != lhs.AggType {
+		if common.Sum != lhs.AggType() {
 			err = utils.StackError(nil, "LHS of avg node must be sum node")
 			return
 		}
-		rhs, ok := mn.children[0].(*MergeNode)
+		rhs, ok := mn.children[1].(common.MergeNode)
 		if !ok {
 			err = utils.StackError(nil, "RHS of avg node must be count node")
 			return
 		}
-		if common.Count != rhs.AggType {
+		if common.Count != rhs.AggType() {
 			err = utils.StackError(nil, "RHS of avg node must be count node")
 			return
 		}
 	}
 
 	childrenResult := make([]queryCom.AQLQueryResult, nChildren)
-	errs := make([]error, nChildren)
+	errs := 0
```

Comment:
better name it as nerrs, errs sounds like an array of err not a count

Reply:
addressed